graphium/trainer/predictor.py

```diff
@@ -96,20 +95,46 @@ def __init__(
             )
             eval_options[task].check_metrics_validity()
 
+        # Work-around to retain task level when model_kwargs are modified for FullGraphFinetuningNetwork
+        if "task_heads_kwargs" in model_kwargs.keys():
+            task_heads_kwargs = model_kwargs["task_heads_kwargs"]
+        elif "pretrained_model_kwargs" in model_kwargs.keys():
+            # This covers finetuning cases where we finetune from the task_heads
+            task_heads_kwargs = model_kwargs["pretrained_model_kwargs"]["task_heads_kwargs"]
+        else:
+            raise ValueError("incorrect model_kwargs")
+        self.task_heads_kwargs = task_heads_kwargs
+
         self._eval_options_dict: Dict[str, EvalOptions] = eval_options
         self._eval_options_dict = {
             self._get_task_key(
-                task_level=model_kwargs["task_heads_kwargs"][key]["task_level"], task=key
+                task_level=task_heads_kwargs[key]["task_level"],
+                task=key
+                # task_level=model_kwargs["task_heads_kwargs"][key]["task_level"], task=key
             ): value
             for key, value in self._eval_options_dict.items()
         }
         # Setting the flag options
         self._flag_options = FlagOptions(flag_kwargs=flag_kwargs)
 
         self.model = self._model_options.model_class(**self._model_options.model_kwargs)
+
+        # Maintain module map to easily select modules
+        # We now need to define the module_map in pretrained_model in FinetuningNetwork
+        # self._module_map = OrderedDict(
+        #     pe_encoders=self.model.encoder_manager,
+        #     pre_nn=self.model.pre_nn,
+        #     pre_nn_edges=self.model.pre_nn_edges,
+        #     gnn=self.model.gnn,
+        #     graph_output_nn=self.model.task_heads.graph_output_nn,
+        #     task_heads=self.model.task_heads.task_heads,
+        # )
+
         loss_fun = {
             self._get_task_key(
-                task_level=model_kwargs["task_heads_kwargs"][key]["task_level"], task=key
+                task_level=task_heads_kwargs[key]["task_level"],
+                task=key
```

Comment:
Again, don't like how the config structure is imposed. Perhaps `task_level` should simply be passed to the `PredictorModule` to keep flexibility

Reply:
Good idea, this is implemented in a3d471540bc3f69cc0af2346ab0e66141b65311b